### DISCUSSION TITLE
Use same tftp server for multiple ipxe clients based on IP range

### DISCUSSION THREAD

**Discussion by parmarnandish** [ipxe/ipxe#362] [Q&A]
**Intro**

Hi, I am trying to use a centos box for dhcp and tftp function, for ipxe boot. 
I have one subnet configured for dhcp, but it has multiple ranges and each range has a different ipxe boot url. 

**Problem** 

Problem is that once the hosts boot into ipxe, it only takes the URL from the range that is at the very bottom, regardless of which range the dhcp ip is assigned from - https://10.10.2.12:6501/vmw/rbd/tramp

For e.g., my host with 10.10.2.212 should get this ipxe url -https://10.10.2.12:6501/vmw/rbd/tramp
and my host with IP 10.10.2.52 should get this ipxe url - https://10.10.2.13:6501/vmw/rbd/tramp

but they are both ipxe booting from this URL https://10.10.2.12:6501/vmw/rbd/tramp

I have also tried to define pool, but the filename option inside the pool does not take effect. 

**Desired results**

my host with 10.10.2.212 should get this ipxe url -https://10.10.2.12:6501/vmw/rbd/tramp
and my host with IP 10.10.2.52 should get this ipxe url - https://10.10.2.13:6501/vmw/rbd/tramp

**dhcpd.conf file**
```
#
# DHCP Server Configuration file.
#   see /usr/share/doc/dhcp*/dhcpd.conf.example
#   see dhcpd.conf(5) man page
#
# Domain name

option domain-name "abc.xyz";

# DNS server ips
option domain-name-servers 8.8.8.8;
# Default lease time
default-lease-time 2592000;
# Max lease time
max-lease-time 5184000;
# Log method
log-facility local7;
authoritative;


# Subnet and ip ranges
# ESX ESXi_Mgmt subnet
subnet 10.10.2.0 netmask 255.255.254.0 {


range 10.10.2.52 10.10.2.211;
if ((exists user-class) and (option user-class = "iPXE")) {
filename "https://10.10.2.13:6501/vmw/rbd/tramp";
} else {
filename "undionly.kpxe";
}

range 10.10.2.212 10.10.3.117;
if ((exists user-class) and (option user-class = "iPXE")) {
filename "https://10.10.2.12:6501/vmw/rbd/tramp";
} else {
filename "undionly.kpxe";
}

option subnet-mask 255.255.254.0;
option broadcast-address 10.121.3.255;
option routers 10.121.2.1;
}


# Include reservations
include "/etc/dhcp/reservations.d/domain1.p.opensource";
include "/etc/dhcp/reservations.d/domain1.np.opensource";

**Comment by mcb30** [comment]
From the `dhcpd.conf` manpage:

> When a client is to be booted, its boot parameters are determined by consulting that client's host  declaration
> (if  any),  and  then  consulting  any class declarations matching the client, followed by the pool, subnet and
> shared-network declarations for the IP address assigned to the client.  Each of these declarations  itself 
> appears  within  a  lexical  scope,  and  all declarations at less specific lexical scopes are also consulted for
> client option declarations.  Scopes are never considered twice, and if parameters are declared in more than one
> scope, the parameter declared in the most specific scope is the one that is used.

Your current configuration places both `if` blocks in the same scope (since a `range` declaration does not declare any kind of scope).  I suspect you will need to get `pool` working.

**Reply by parmarnandish** [reply]
Thank you @mcb30. Looks like Class declaration takes precedence before Pool declaration. If all my classes are ipxe, then declaring in pools would probably not work as well. Any other ideas to make this work?
